src/artemis/nexus_writing/write_nexus.py

```diff
@@ -48,6 +49,35 @@
 }
 
 
+def create_parameters_for_first_file(parameters: FullParameters):
+    new_params = deepcopy(parameters)
+    new_params.grid_scan_params.z_axis = GridAxis(
+        parameters.grid_scan_params.z1_start, 0, 0
+    )
+    new_params.detector_params.num_images = (
+        parameters.grid_scan_params.x_steps * parameters.grid_scan_params.y_steps
+    )
```

Comment:
Does this need setting in the 2nd file params method as well? if y/z steps are different the 2 files should have different numbers of images

Reply:
No, the second nexus file needs to know the full number of images that have been taken, which is what will be in `new_params.detector_params.num_images` if we don't override it. Effectively this value stores the upper limit for the VDS slicing. It's messy, I agree, really needs a second refactor but think we'll do it in https://github.com/DiamondLightSource/python-artemis/issues/164